src/Generator/Generators/CLI/CLIMarshal.cs

```diff
@@ -40,13 +40,15 @@ public override bool VisitTagType(TagType tag, TypeQualifiers quals)
             return decl.Visit(this);
         }
 
+        private static int arrayNum = 0;
```

Comment:
Thanks. I implemented something similar in CSharpTextTemplate but could not do same here. I will look into ParameterIndex and try to implement.